banzai_nres/qc/qc_wavelength.py

```diff
@@ -1,55 +1,78 @@
-        x_diff_Dlambda, order_diff_Dlambda = self.calculate_2d_metrics(image, Delta_lambda)
-        qc_results = {'sigma_Dlambda': sigma_Dlambda, 'good_sigma_Dlambda': good_sigma_Dlambda,
-                      'raw_chi_squared': raw_chi_squared, 'good_chi_squared': good_chi_squared,
-                      'Delta_lambda': Delta_lambda, 'x_diff_Dlambda': x_diff_Dlambda,
-                      'order_diff_Dlambda': order_diff_Dlambda}
+        dlambda = self.calculate_delta_lambda(image, lab_lines)
+        result = self.calculate_1d_metrics(image, dlambda, lab_lines, dlambda_match_threshold)
+        sigma_dlambda, matched_sigma_dlambda, chi2, matched_chi2, num_matched_lines, velocity_precision = result
+        qc_results = {'SIGLAM': np.round(matched_sigma_dlambda, 4),
+                      'PRECISN': np.round(velocity_precision.to(units.meter/units.second).value, 4),
+                      'CHISQ': np.round(matched_chi2, 4),
+                      'LINENUM': len(image.features['wavelength']),
+                      'MLINENUM': num_matched_lines}
+        qc_description = {'SIGLAM': 'wavecal residuals Angstroms',
+                          'PRECISN': 'm/s wavecal precision',
+                          'CHISQ': 'chisquared goodness of wavecal fit',
+                          'LINENUM': 'Number of lines found on detector',
+                          'MLINENUM': 'Number of matched lines'}
         qc.save_qc_results(self.runtime_context, qc_results, image)
+        # saving the results to the image header
+        for key in qc_results.keys():
+            image.meta[key] = (qc_results[key], qc_description[key])
+
+        # print the most easily understood metric to the log
+        logger.info(f'wavecal precision (m/s) = {qc_results["PRECISN"]}', image=image)
+        if qc_results['PRECISN'] > 10 or qc_results['PRECISN'] < 3:
+            logger.warning(f' Final calibration precision is outside the expected range '
+                           f'wavecal precision (m/s) = '
+                           f'{qc_results["PRECISN"]}', image=image)
         return image
 
     def calculate_delta_lambda(self, image, lab_lines):
+        """
+        :param image:
+        :param lab_lines: list of laboratory wavelengths
+        :return: Dlambda: ndarray. The residual between the measured wavelength and the laboratory wavelength
+
+        """
         measured_wavelengths = image.features['wavelength']
-        Delta_lambda = measured_wavelengths - lab_lines
-        return Delta_lambda
+        Dlambda = measured_wavelengths - lab_lines
+        return Dlambda
 
-    def calculate_1d_metrics(self, image, Delta_lambda):
-        sigma_Dlambda = np.std(Delta_lambda)
-        low_scatter_lines = np.isclose(Delta_lambda, 0, atol=0.1)
-        good_sigma_Dlambda = np.std(Delta_lambda[low_scatter_lines])
+    def calculate_1d_metrics(self, image, delta_lambda, lab_lines, dlambda_match_threshold=0.1):
+        sigma_dlambda = np.std(delta_lambda)
```

Comment:
This might get covered in the documentation better, but maybe add some comments on why we are doing these checks.